nft/payment_obligation.go

```diff
@@ -54,10 +54,22 @@ func (t TokenID) String() string {
 	return hexutil.Encode(t[:])
 }
 
+// MintNFTRequest holds required fields for minting NFT
+type MintNFTRequest struct {
+	DocumentID               []byte
+	RegistryAddress          string
+	DepositAddress           string
+	ProofFields              []string
+	GrantNFTReadAccess       bool
+	SubmitTokenProof         bool
+	SubmitRoleProof          string
```

Comment:
I would rename this to make it clear that this is the actual role. And as mentioned above, I'd make it a []byte
```suggestion
	AccountRoleID          []byte
```

Reply:
This is client facing API request, we cannot have []byte. We will take it as a hex if this user gives the 32 byte role label

Reply:
Ah didn't catch that this is the client model. Somewhere else I spotted string as well.